Disable Button when Invalid email being typed in

enpuyou opened the pull request:
<!-- What's this PR for?  (Just a link to an issue is fine.) -->
#3128 

**Testing Plan:** <!-- How have you tested? -->

Tested from the development environment

**GIFs or Screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->


<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

enpuyou commented on templates/zerver/find_account.html:
```diff
@@ -37,10 +37,31 @@ <h1 class="get-started">{{ _("Find your Zulip accounts") }}…</h1>
+                            var a = document.getElementById('emails');
+                            var b = document.getElementById('output');
+                            var button = document.getElementById('submitButton');
+                            var re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```
Added in the latest commit

enpuyou commented on templates/zerver/find_account.html:
```diff
@@ -37,10 +37,31 @@ <h1 class="get-started">{{ _("Find your Zulip accounts") }}…</h1>
                     <div class="input-box moving-label horizontal">
                         <div class="inline-block relative">
-                            <input type="text" autofocus id="emails" name="emails" required />
+                            <input type="text" autofocus id="emails" name="emails" onkeydown="test();" required />
```
After trying with `onchange` and `onkeydown`, I'm currently switching to `oninput` that will  handle auto-fill or paste event

enpuyou commented on templates/zerver/find_account.html:
```diff
@@ -37,10 +37,31 @@ <h1 class="get-started">{{ _("Find your Zulip accounts") }}…</h1>
+                              document.getElementById("submitButton").disabled = false;
+                            }
+                          }
+                        </script>
```
This is being fixed in the latest commit

enpuyou commented on templates/zerver/find_account.html:
```diff
@@ -37,10 +37,31 @@ <h1 class="get-started">{{ _("Find your Zulip accounts") }}…</h1>
+                            }
+                          }
+                        </script>
+                        <a id="output">Enter email<a>
```
Fixed

timabbott commented:
@enpuyou thanks for working on this.  Can you clean up the Git history?  This PR has ~50 commits!  

Check out our [GitHub guide](https://zulip.readthedocs.io/en/latest/git-guide.html) and [commit guidelines](https://zulip.readthedocs.io/en/latest/contributing/version-control.html) for more details.

enpuyou commented:
@timabbott Thanks for checking on this! I have squashed the commits into one single commit since the task is just to disable a button while any invalid email is being typed in. We changed the original display text `Find accounts` to `Find teams`, as we found the original text to be a little bit confusing, especially when compared with the issue body in #3128 . I certainly believe that this might not be appropriate since all the translations are based on `Find accounts`. We can certainly make some further commits on better resolving this issue.
